thunx-spring/src/main/java/com/contentgrid/thunx/spring/data/rest/AbacRepositoryInvokerAdapter.java

```diff
@@ -147,9 +146,25 @@ public <T> T invokeSave(T object) {
                 status = transactionManager.getTransaction(new DefaultTransactionDefinition());
             }
 
+            // when object has no id, there is no pre-save-check required
+            // when object has an 'id', do:
+            //   1. invokeFindById without predicate
+            //   2. invokeFindById with predicate
+            //   case a) - both (1) and (2) are NOT present: create new -> check afterwards the new object was allowed
+            //   case b) - both (1) and (2) are present: update -> check afterwards the updates are allowed
+            //   case c) - if (1) is NOT present, but (2) is: impossible case
+            //   case d) - if (1) is present, but (2) is NOT: permission denied
+            this.idFunction.apply(object).ifPresent(id -> {
+                if (super.invokeFindById(id) /* without predicate */.isPresent()) {
+                    if (this.invokeFindById(id) /* with predicate */.isEmpty()) {
+                        throw new ResourceNotFoundException(String.format("id: %s", id));
```

Comment:
Acknowledged that this is not an access-denied-exception (http 403), but results in an http 404 currently.

A follow up PR will introduce a custom access-denied-exception, that gets translated into an http 403